ocs_ci/ocs/resources/mcg.py

```diff
@@ -272,3 +296,289 @@ def _retrieve_reduction_data():
                 'Not enough data reduction. Something is wrong.'
             )
             return False
+
+    def request_aws_credentials(self):
+        """
+        Uses a CredentialsRequest CR to create an AWS IAM that allows the program
+        to interact with S3
+
+        Returns:
+            OCS: The CredentialsRequest resource
+        """
+        awscreds_data = templating.load_yaml(constants.MCG_AWS_CREDS_YAML)
+        req_name = create_unique_resource_name('awscredreq', 'credentialsrequests')
+        awscreds_data['metadata']['name'] = req_name
+        awscreds_data['metadata']['namespace'] = self.namespace
+        awscreds_data['spec']['secretRef']['name'] = req_name
+        awscreds_data['spec']['secretRef']['namespace'] = self.namespace
+        creds_request = create_resource(**awscreds_data)
+
+        def _retrieve_credreq_provision_state():
+            return run_cmd(
+                f"oc get credentialsrequests {creds_request.name} -o jsonpath={{.status.provisioned}}"
+            )
+
+        try:
+            for provisioned_state in TimeoutSampler(30, 5, _retrieve_credreq_provision_state):
+                if provisioned_state == "true":
+                    sleep(10)
+                    logger.info(f'Secret created successfully.')
+                    break
+                else:
+                    logger.info(
+                        f'Credentials were not yet provisioned or could not be found. '
+                        f'Retrying...'
+                    )
+        except TimeoutExpiredError:
+            logger.error(
+                'Failed to create credentials'
+            )
+
+        return creds_request
+
+    def create_new_aws_connection(self, conn_name=None):
+        """
+        Creates a new NooBaa connection to an AWS backend
+
+        Args:
+            conn_name: The connection name to be used
+
+        Returns:
+            bool: False if the connection creation failed
+
+        """
+        if conn_name is None:
+            conn_name = create_unique_resource_name('awsconnection', 'mcgconn')
+
+        params = {
+            "auth_method": "AWS_V4",
+            "endpoint": "https://s3.amazonaws.com",
+            "endpoint_type": "AWS",
+            "identity": self.aws_access_key_id,
+            "name": conn_name,
+            "secret": self.aws_access_key
+        }
+
+        try:
+            for resp in TimeoutSampler(
+                30, 3, self.send_rpc_query, 'account_api', 'add_external_connection', params
+            ):
+                if 'error' not in resp.text:
+                    logger.info(f'Connection {conn_name} created successfully')
```

Comment:
this function returns False only. Please adjust it to return True instead of breaking